client/state/sites/domains/assembler.js

```diff
@@ -17,6 +17,20 @@ function assembleGoogleAppsSubscription( googleAppsSubscription ) {
 	return mapKeys( googleAppsSubscription, ( value, key ) => camelCase( key ) );
 }
 
+function assembleCurrentUserCannotAddEmailReason( reason ) {
+	if ( ! reason || ! reason.errors ) {
+		return null;
+	}
+	const errorDetails = Object.entries( reason.errors ).map( ( entry ) => {
+		const [ errorCode, errorMessages ] = entry;
+		return {
+			code: errorCode,
+			message: errorMessages[ 0 ],
+		};
+	} );
+	return errorDetails[ 0 ];
```

Comment:
Seems like it might be nice to ascertain that `errorDetails` actually contains at least an element. Furthermore, why do we need to map all the entries if we just need the first?

Reply:
So the mapping here is because `reason.errors` is an object not an array, with the error codes as the field names -- we don't know ahead of time which error(s) we're going to get back.

I'll add the check for an empty array, but I don't know that we can avoid iterating over the error keys.

Reply:
I was thinking `Object.entries` returns an array, and we could just grab the first one if it exists, but it's more of a personal preference.